### PR TITLE
Anomaly parameters exposed from OPF/swarming

### LINKED ISSUE #1305: Expose Anomaly's parameters to OPF, so it can be used in swarming
update the `OPFClient` (or which file it is?) so the parameters to Anomaly class are exposed for swarming-over. For example `slidingWindowSize` or `Anomaly.MODE_*`
Modify anomaly swarming example to use them as test/example.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/4365964-expose-anomaly-s-parameters-to-opf-so-it-can-be-used-in-swarming?utm_campaign=plugin&utm_content=tracker%2F290115&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F290115&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by breznak**
This change allows us to set anomaly params in OPF description file (and thus swarm over them), eg: 
` 'anomalyParams': {
       'mode': 'pure', # pure(=default) / weighted / likelihood
       'slidingWindowSize': 5, # >=0 / None
    },
`

Fixes #1305 
~~Blocked by: https://github.com/numenta/nupic/pull/1353 (builds upon it)~~

### CODE REVIEW

File: nupic/frameworks/opf/clamodel.py [hunk 1]
```diff
@@ -189,7 +187,10 @@ def __init__(self,
     self._predictedFieldName = None
     self._numFields = None
     # init anomaly
-    self._anomalyInst = Anomaly()
+    windowSize = anomalyParams.get('slidingWindowSize', None)
+    mode = anomalyParams.get('mode', 'pure')
+    self._anomalyInst = Anomaly(slidingWindowSize=windowSize, mode=mode)
+    #FIXME self.__logger.debug(self._anomalyInst)
```
Comment (scottpurdy) [inline nupic/frameworks/opf/clamodel.py]: Remove this?

File: nupic/frameworks/opf/clamodel.py [hunk 2]
```diff
@@ -189,7 +187,10 @@ def __init__(self,
     self._predictedFieldName = None
     self._numFields = None
     # init anomaly
-    self._anomalyInst = Anomaly()
+    windowSize = anomalyParams.get('slidingWindowSize', None)
```
Comment (scottpurdy) [inline nupic/frameworks/opf/clamodel.py]: Double quotes

File: nupic/frameworks/opf/clamodel.py [hunk 3]
```diff
@@ -219,6 +220,8 @@ def __init__(self,
 
     self.__logger.debug("Instantiated %s" % self.__class__.__name__)
 
+    self._in = None # input data
```
Comment (scottpurdy) [inline nupic/frameworks/opf/clamodel.py]: More desciptive please. Perhaps `self._input`? And then you don't need the comment?

**Comment by scottpurdy** [comment]
Done with review

**Comment by breznak** [comment]
Thanks Scott! 
@rhyolight ready :)

**Pull request merged**